### PR TITLE
VZ-8780: Thanos updates

### PR DESCRIPTION
**Pull request by MelwynPaul**
- New topic in Monitor section
- Updates to Access Verrazzano section

### CODE REVIEW

File: content/en/docs/monitoring/thanos.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+- Long-term metrics retention
+- High availability
+- Easy backup for metrics
+- Efficient data access
+
+For more information on Thanos, see the [Thanos website](https://thanos.io/).
+
+## Components
+
+Thanos comprises of the  following components:
```
Comment (davisgomes) [inline content/en/docs/monitoring/thanos.md:21]: extra space here.

File: content/en/docs/monitoring/thanos.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
+---
+title: "Thanos"
+linkTitle: Thanos
+description: "Learn how to use Thanos to monitor Verrazzano"
+weight: 1
+draft: false
+---
+
+Thanos is a group of components that seamlessly integrates with Prometheus to monitor your applications. You can install and configure Thanos components with Verrazzano and use long-term storage to store metrics. Then, use the Thanos console to query for metrics across all Prometheus instances and long-term storage. Thanos also makes it easier to scale Prometheus horizontally and obtain a global view of data from multiple Prometheus servers.
```
Comment (davisgomes) [inline content/en/docs/monitoring/thanos.md:9]: I'm not sure I'm qualified enough to comment on English here but "Tean, use the Thanos console to..." sounds a bit clunky to me. Maybe something like "The Thanos console enables you to ..." would suit this better.

File: content/en/docs/monitoring/thanos.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+## Components
+
+Thanos comprises of the  following components:
+
+| Components     | Description                                                                                                                                         |
+|----------------|-----------------------------------------------------------------------------------------------------------------------------------------------------|
+| Sidecar        | Container that resides in the Prometheus pod. It connects to Prometheus, reads its data for queries, and uploads it to long-term storage.           |
+| Store Gateway  | Serves metrics from long-term storage.                                                                                                              |
+| Query          | Implements Prometheus API to aggregate data from the underlying components and provides a user interface for querying across all Prometheus stores. |
+| Query Frontend | Implements Prometheus API and proxies it to Query while caching the response and optionally splits queries .                                        |
```
Comment (davisgomes) [inline content/en/docs/monitoring/thanos.md:28]: extra space before period

File: content/en/docs/monitoring/thanos.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+## Enable Thanos
+
+You can enable the Thanos Prometheus sidecar, Query, and Query Frontend components.
```
Comment (davisgomes) [inline content/en/docs/monitoring/thanos.md:32]: Should sidecar be capitalized?

File: content/en/docs/monitoring/thanos.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+```
+
+</div>
+{{< /clipboard >}}
+
+#### Step 2: Create a secret
+
+Create the secret for object storage configuration.
+
+The following example uses the filename `objstore.yml` so that the key name in the secret is exactly what is expected by the Store Gateway.
```
Comment (davisgomes) [inline content/en/docs/monitoring/thanos.md:61]: Maybe consider making this a note and a bit more suggestive that any other file name will not work.

File: content/en/docs/monitoring/thanos.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+                  name: objstore-config
+                  key: objstore.yml
+    thanos:
+      enabled: true
+```
+
+</div>
+{{< /clipboard >}}
+
+## Configure Thanos Store Gateway
```
Comment (davisgomes) [inline content/en/docs/monitoring/thanos.md:104]: Should this be step 4?

File: content/en/docs/monitoring/thanos.md (hunk 1 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Fixed. Thank you.

File: content/en/docs/monitoring/thanos.md (hunk 4 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md:32]: Yes, I think so.
I see the Thanos components are all capitalized.

File: content/en/docs/monitoring/thanos.md (hunk 2 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Thank you, Davis.
Wanted it to be a continuation after you install and configure. 
Updated now.

File: content/en/docs/monitoring/thanos.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+#### Step 2: Create a secret
+
+Create the secret for object storage configuration.
+
+**NOTE**: Use the `objstore.yml` filename to match the key name in the secret.
+
+{{< clipboard >}}
+<div class="highlight">
+
+```
```
Comment (fredtibbitts) [inline content/en/docs/monitoring/thanos.md:66]: The namespace needs to exist before creating the secret, so add:
```
kubectl create namespace verrazzano-monitoring
```

File: content/en/docs/monitoring/thanos.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+k create secret generic -n verrazzano-monitoring objstore-config --from-file objstore.yml
```
Comment (fredtibbitts) [inline content/en/docs/monitoring/thanos.md:67]: "k" -> "kubectl"

File: content/en/docs/monitoring/thanos.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+
+Store Gateway helps you to query metrics in long-term storage.
+
+Enable the `storegateway` with the following Verrazzano configuration, where `objstore-config` is an existing secret.
+
+{{< clipboard >}}
+<div class="highlight">
+
+```
+spec:
```
Comment (fredtibbitts) [inline content/en/docs/monitoring/thanos.md:114]: Should we just combine this spec snippet with the one above and show the user how to enable store gateway at the same time? I think there are pros and cons to splitting it out like this, I'm not sure which way is better, just asking the question.

File: content/en/docs/monitoring/thanos.md (hunk 5 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Thank you, Davis.

File: content/en/docs/monitoring/thanos.md (hunk 7 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md:67]: Thank you. 
Updated

File: content/en/docs/monitoring/thanos.md (hunk 8 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Thank you. 
Updated

File: content/en/docs/monitoring/thanos.md (hunk 6 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md:106]: It was step 4.
But, we thought we can have a separate section as 'Configure Thanos Store Gateway'. 
Also, Fred has a question about combining step 3 and 'Configure Thanos Store Gateway'.

File: content/en/docs/monitoring/thanos.md (hunk 9 above)
Comment (davisgomes) [reply content/en/docs/monitoring/thanos.md:116]: I was thinking the same thing. It avoids the user having to do two upgrades to enable all Thanos

File: content/en/docs/monitoring/thanos.md [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+metadata:
+  name: my-verrazzano
+spec:
+  components:
+    prometheusOperator:
+      enabled: true
+      overrides:
+      - values:
+          prometheus:
+            thanos
```
Comment (davisgomes) [inline content/en/docs/monitoring/thanos.md:92]: missing colon here

Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Thank you. 
Content updated.

File: content/en/docs/monitoring/thanos.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+
+{{< clipboard >}}
+<div class="highlight">
+
+```
+type: OCI
+config:
+  provider: "raw"
+  bucket: "<bucket_name>"
+  compartment_ocid: "<stack_compartment_ocid>"
```
Comment (desagar) [inline content/en/docs/monitoring/thanos.md:46]: what is "stack_compartment_ocid"? I think this should be "<bucket compartment OCID>".

Also, everything inside the angle brackets is kind of explanatory, so using underscores like "tenancy_ocid" is not needed. It should probably read "tenancy OCID". Likewise for others.

File: content/en/docs/monitoring/thanos.md (hunk 9 above)
Comment (desagar) [reply content/en/docs/monitoring/thanos.md:116]: I agree that combining them would be nice. However, it does make it more difficult for the user to know what a minimal configuration is, so would require a note some kind of note.

I'm wondering based on Mike's previous comments about making the easy thing obvious, we should have another much smaller snippet first, showing the minimal configuration without external storage (for testing purposes).

File: content/en/docs/monitoring/thanos.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+
+</div>
+{{< /clipboard >}}
+
+#### Step 2: Create a secret
+
+Create the secret for object storage configuration.
+
+The following example uses the filename `objstore.yml`. Using the `objstore.yml` filename helps you to generate the right key when creating the bucket secret.
+The Store Gateway requires the secret key `objstore.yml` to fetch the bucket credentials.
```
Comment (desagar) [inline content/en/docs/monitoring/thanos.md:62]: I know you made this a little more prominent than it was before, but I still think there's a bit of room for change here. Using that filename is optional - the user just has to make sure that the key is exactly what Store Gateway expects.

I'd reword to something like the following (or equivalent that you come up with):
The following example uses the filename `objstore.yml`. The Thanos Store Gateway requires the key in the secret to be `objstore.yml`. If you use a different filename, make sure that the key in the secret is exactly `objstore.yml`.

File: content/en/docs/monitoring/thanos.md (hunk 9 above)
Comment (fredtibbitts) [reply content/en/docs/monitoring/thanos.md:116]: @MelwynPaul ok I think we've decided we want a very simple first example CR that does not include storage, so this:
```
apiVersion: install.verrazzano.io/v1beta1
kind: Verrazzano
metadata:
  name: my-verrazzano
spec:
  components:
    prometheusOperator:
      enabled: true
      overrides:
      - values:
          prometheus:
            thanos:
              integration: sidecar
    thanos:
      enabled: true
```
That should be the very first, simple example. That will enable the Prometheus Thanos Sidecar, Thanos Query, and Thanos Query Frontend.

Then add the instructions for enabling storage: create the secret, etc. and provide this as the example CR:
```
apiVersion: install.verrazzano.io/v1beta1
kind: Verrazzano
metadata:
  name: my-verrazzano
spec:
  components:
    prometheusOperator:
      enabled: true
      overrides:
      - values:
          prometheus:
            thanos:
              integration: sidecar
            prometheusSpec:
              thanos:
                objectStorageConfig:
                  name: objstore-config
                  key: objstore.yml
    thanos:
      enabled: true
      overrides:
      - values:
          existingObjstoreSecret: objstore-config
          storegateway:
            enabled: true
```
That additionally enables Thanos Store Gateway and configures the Thanos Sidecar to write to object storage and the Store Gateway to read from object storage.

File: content/en/docs/monitoring/thanos.md [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+- Long-term metrics retention
+- High availability
+- Easy backup for metrics
+- Efficient data access
+
+For more information on Thanos, see the [Thanos website](https://thanos.io/).
+
+## Components
+
+Thanos comprises of the following components:
```
Comment (desagar) [inline content/en/docs/monitoring/thanos.md:21]: Recommend changing "Thanos comprises of the following components" to "Verrazzano supports the following Thanos components"

Thanos has several other components that Verrazzano doesn't currently support.

File: content/en/docs/monitoring/thanos.md (hunk 11 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Thank you, Deva. 
Content updated. Please verify.

File: content/en/docs/monitoring/thanos.md (hunk 12 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Thank you. 
Content updated.

File: content/en/docs/monitoring/thanos.md (hunk 13 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Yes. Thank you. 
Content updated.

File: content/en/docs/monitoring/thanos.md (hunk 9 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md:114]: Thank you, @fredtibbitts. 
Updated content. Please verify.
Modified the 'Thanos Store Gateway' as Step 4.

File: content/en/docs/monitoring/thanos.md [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+            thanos:
+              integration: sidecar
+    thanos:
+      enabled: true
+```
+
+</div>
+{{< /clipboard >}}
+
+#### Step 4: Enable storage and Thanos Store Gateway
```
Comment (desagar) [inline content/en/docs/monitoring/thanos.md:100]: Rather than being a Step 4 in the series, this is an optional "advanced" configuration. At a minimum I'd call this "Optional Step 4: Enable storage and Thanos Store Gateway", which the user can do if they want to use long term storage.

File: content/en/docs/monitoring/thanos.md [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+| Components     | Description                                                                                                                                         |
+|----------------|-----------------------------------------------------------------------------------------------------------------------------------------------------|
+| Sidecar        | Container that resides in the Prometheus pod. It connects to Prometheus, reads its data for queries, and uploads it to long-term storage.           |
+| Store Gateway  | Serves metrics from long-term storage.                                                                                                              |
+| Query          | Implements Prometheus API to aggregate data from the underlying components and provides a user interface for querying across all Prometheus stores. |
+| Query Frontend | Implements Prometheus API and proxies it to Query while caching the response and optionally splits queries.                                        |
+
+## Enable Thanos
+
+You can enable the Thanos Prometheus Sidecar, Query, and Query Frontend components.
```
Comment (rosemarymarano) [inline content/en/docs/monitoring/thanos.md:32]: -> To enable the Thanos Prometheus Sidecar, Query, and Query Frontend components:

File: content/en/docs/monitoring/thanos.md [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+
+### Step 1: Create a YAML configuration file
+
+Create an `objstore.yml` file using the OCI object storage.
```
Comment (rosemarymarano) [inline content/en/docs/monitoring/thanos.md:36]: -> using OCI object storage.

File: content/en/docs/monitoring/thanos.md [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+</div>
+{{< /clipboard >}}
+
+#### Step 2: Create a secret
+
+Create the secret for object storage configuration.
+
+The following example uses the filename `objstore.yml`. The Thanos Store Gateway requires the key in the secret to be `objstore.yml`. If you use a different filename, make sure that the key in the secret is exactly `objstore.yml`.
```
Comment (rosemarymarano) [inline content/en/docs/monitoring/thanos.md:61]: -> file name (two words, fix globally)

File: content/en/docs/monitoring/thanos.md [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+      - values:
+          existingObjstoreSecret: objstore-config
+          storegateway:
+            enabled: true
+```
+
+</div>
+{{< /clipboard >}}
+
+## Access Thanos Query Console
```
Comment (rosemarymarano) [inline content/en/docs/monitoring/thanos.md:140]: -> Access the Thanos Query console

File: content/en/docs/monitoring/thanos.md [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+
+You can access the Thanos Query console using the instructions at [Get the consoles URLs]({{< relref "/docs/access/#get-the-consoles-urls" >}}). section in [Access Verrazzano]({{< relref "/docs/access/" >}}).
```
Comment (rosemarymarano) [inline content/en/docs/monitoring/thanos.md:142]: -> . (period is a typo, unless you end the sentence here, which is what I would suggest.) 
-> section in [Access Verrazzano]({{< relref "/docs/access/" >}}). (Don't need this part of the sentence. When they go to the x-ref, they are in the Access Verrazzano doc.)

File: content/en/docs/monitoring/thanos.md [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+
+Advantages of using Thanos:
+- Long-term metrics retention
+- High availability
+- Easy backup for metrics
+- Efficient data access
+
+For more information on Thanos, see the [Thanos website](https://thanos.io/).
+
+## Components
```
Comment (rosemarymarano) [inline content/en/docs/monitoring/thanos.md:19]: -> Thanos components

**Review by rosemarymarano — COMMENTED** [review]
minor edits and suggestions

File: content/en/docs/monitoring/thanos.md (hunk 16 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Thank you.
Updated content.

File: content/en/docs/monitoring/thanos.md (hunk 18 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Thank you.
Updated content.

File: content/en/docs/monitoring/thanos.md (hunk 19 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Yes, agreed.

File: content/en/docs/monitoring/thanos.md (hunk 20 above)
Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Thank you.
Updated content.

**Review by rosemarymarano — APPROVED** [review]

**Review by desagar — APPROVED** [review]

File: content/en/docs/monitoring/thanos.md [hunk 21]
```diff
@@ -0,0 +1,142 @@
...
+
+```
+$ kubectl create namespace verrazzano-monitoring
+$ kubectl create secret generic -n verrazzano-monitoring objstore-config --from-file objstore.yml
+```
+
+</div>
+{{< /clipboard >}}
+
+#### Step 3: Enable the Prometheus Thanos Sidecar, Thanos Query, and Thanos Query Frontend
```
Comment (fredtibbitts) [inline content/en/docs/monitoring/thanos.md:74]: Step 1 and 2 are only needed when you get to step 4. This simple CR should be the very first step, and then after should be the config file, secret creation, and CR that shows how to configure long-term storage. Basically move this to step 1 and move everything else down.

Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Thank you, @fredtibbitts 
Moved "Enable the Prometheus Thanos Sidecar, Thanos Query, and Thanos Query Frontend" step to Step 1.

File: content/en/docs/monitoring/thanos.md [hunk 22]
```diff
@@ -0,0 +1,142 @@
...
+            thanos:
+              integration: sidecar
+    thanos:
+      enabled: true
+```
+
+</div>
+{{< /clipboard >}}
+
+### Step 2: Create a YAML configuration file
```
Comment (fredtibbitts) [inline content/en/docs/monitoring/thanos.md:60]: Thanks for moving the CRs around, much better. I think we need to indicate that everything after step 1 is optional, and only needed if you want to enable long-term storage of metrics.

Comment (MelwynPaul) [reply content/en/docs/monitoring/thanos.md]: Thank you, @fredtibbitts 
Added the "Enable long-term storage metrics" section and moved the optional steps under it.
Please check if you like this idea.

**Review by fredtibbitts — APPROVED** [review]
LGTM, thanks for making all of the updates

File: content/en/docs/access/_index.md [hunk 23]
```diff
@@ -118,6 +120,7 @@ The following is an example of the output:
 "opensearchDashboardsUrl": "https://opensearchDashboards.vmi.system.default.11.22.33.44.nip.io",
 "prometheusUrl": "https://prometheus.vmi.system.default.11.22.33.44.nip.io",
 "rancherUrl": "https://rancher.default.11.22.33.44.nip.io"
+"thanosQueryUrl": "https://thanos-query.default.172.18.0.231.nip.io"
```
Comment (desagar) [inline content/en/docs/access/_index.md:123]: We use 11.22.33.44 for all the URLs - thanosQueryUrl should also use that.

File: content/en/docs/access/_index.md [hunk 24]
```diff
@@ -47,6 +47,7 @@ Verrazzano Status
     openSearchUrl: https://opensearch.vmi.system.default.172.18.0.231.nip.io
     prometheusUrl: https://prometheus.vmi.system.default.172.18.0.231.nip.io
     rancherUrl: https://rancher.default.172.18.0.231.nip.io
+    thanosQueryUrl: https://thanos-query.default.172.18.0.231.nip.io
```
Comment (desagar) [inline content/en/docs/access/_index.md:50]: I think perhaps we should take this opportunity to change all of these as well to 11.22.33.44

File: content/en/docs/access/_index.md (hunk 23 above)
Comment (MelwynPaul) [reply content/en/docs/access/_index.md]: Thank you. 
Updated thanosQueryUrl and argoCD URL to 11.22.33..44

File: content/en/docs/access/_index.md (hunk 24 above)
Comment (MelwynPaul) [reply content/en/docs/access/_index.md]: Thank you. 
Updated all URLs to 11.22.33..44
